src/common/hooks/account/use-account-transactions-with-transfers.hooks.ts

```diff
@@ -0,0 +1,43 @@
+import { useQuery, UseQueryOptions } from 'react-query';
+import { useCurrentAccountStxAddressState } from '@store/accounts/account.hooks';
+import { useCurrentNetworkState } from '@store/network/networks.hooks';
+import { useAccountsApi } from '@store/common/api-clients.hooks';
+import { AddressTransactionWithTransfers } from '@stacks/stacks-blockchain-api-types';
+import { useAccountTransactionsWithTransferState } from '@store/accounts/transactions-with-transfer.hooks';
+import { DEFAULT_LIST_LIMIT, QueryRefreshRates } from '@common/constants';
+import { PaginatedResults } from '@common/types';
+
+const QUERY_OPTIONS = {
+  refetchInterval: QueryRefreshRates.MEDIUM,
+  refetchOnMount: 'always',
+  refetchOnReconnect: 'always',
+  refetchOnWindowFocus: 'always',
+};
+
+export function useGetAccountTransactionsWithTransfer(reactQueryOptions: UseQueryOptions = {}) {
+  const principal = useCurrentAccountStxAddressState();
+  const networkUrl = useCurrentNetworkState().url;
+  const { accountsApi } = useAccountsApi();
+  const fetch = () => {
+    if (!principal) return;
+    return accountsApi.getAccountTransactionsWithTransfers({
+      principal,
+      limit: DEFAULT_LIST_LIMIT,
+    });
+  };
+
+  return useQuery(['accountNonce', principal, networkUrl], fetch, {
+    enabled: !!principal || !!networkUrl,
+    ...reactQueryOptions,
+  });
+}
+
+export function useAccountTransactionsWithTransfer() {
```

Comment:
Should this always be plural throughout? `useAccountTransactionsWithTransfers`

Reply:
This is plural, so makes sense to me to keep it plural `accountsApi.getAccountTransactionsWithTransfers`?